File: sdks/python/examples/basic_chat.py
"""
Basic chat example.

Usage:
    ANTHROPIC_API_KEY=sk-ant-... python examples/basic_chat.py
    OPENAI_API_KEY=sk-...       python examples/basic_chat.py openai
"""
import asyncio
import sys
from motosan_ai import Client

async def main():
    provider = sys.argv[1] if len(sys.argv) > 1 else "anthropic"

    client = Client(provider=provider)
    response = await client.chat([
        {"role": "user", "content": "What is the capital of France? Answer in one sentence."}
    ])

    print(f"Response: {response.content}")
    print(f"Model: {response.model}")
    print(f"Tokens: {response.usage.input_tokens} in, {response.usage.output_tokens} out")

asyncio.run(main())

File: sdks/rust/examples/basic_chat.rs
//! Basic chat example using Anthropic.
//!
//! Run with:
//!   ANTHROPIC_API_KEY=sk-ant-... cargo run --example basic_chat --features anthropic

use motosan_ai::{Client, Message, Provider};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let api_key = std::env::var("ANTHROPIC_API_KEY")
        .expect("ANTHROPIC_API_KEY environment variable not set");

    let client = Client::builder()
        .provider(Provider::Anthropic)
        .api_key(api_key)
        .build()?;

    let response = client.chat(vec![
        Message::user("What is the capital of France? Answer in one sentence."),
    ]).await?;

    println!("Response: {}", response.content);
    println!("Model: {}", response.model);
    println!("Tokens: {} in, {} out", response.usage.input_tokens, response.usage.output_tokens);

    Ok(())
}
